.jenkinsci/doxygen.groovy

```diff
@@ -1,11 +1,23 @@
 #!/usr/bin/env groovy
 
 def doDoxygen() {
-
-    sh """
-        doxygen Doxyfile
-        #rsync docs/doxygen
-    """
+  try {
+    if (env.GIT_LOCAL_BRANCH ==~ /(master|develop)/ || env.CHANGE_BRANCH == 'develop') {
```

Comment:
I'd suggest to move from ==~ to 
```
env.GIT_LOCAL_BRANCH in ["master","develop"]
```
as previous condition is a regex, and e.g. `pre-develop` branch name will also trigger docs build